src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -125,6 +126,38 @@ namespace OpenRCT2::Ui::Windows
         const ScenarioIndexEntry* _highlightedScenario = nullptr;
         ParkPreview _preview;
         BackgroundWorker::Job _previewLoadJob;
+        bool _initialLoadCommand = true;
+
+        void ReportProgress(uint8_t progress)
+        {
+            if (_initialLoadCommand)
+                return;
+
+            if (progress == 0)
+                GetContext()->OpenProgress(STR_CHECKING_SCENARIO_FILES);
+
+            GetContext()->SetProgress(progress, 100, STR_STRING_M_PERCENT);
+
+            if (progress == 100)
+                GetContext()->CloseProgress();
+        }
+
+        void LoadScenarioList()
+        {
+            try
+            {
+                ReportProgress(0);
+                ScenarioRepositoryScan();
+                ReportProgress(100);
+
+                _initialLoadCommand = false;
+            }
+            catch (const std::exception& e)
+            {
+                Console::Error::WriteLine("Error rebuilding scenario index: %s", e.what());
```

Comment:
Surprisingly, `Console::Error::WriteLine` internally calls `Context::WriteLine`. Might as well call that directly; saves having to include `Console.hpp` as well.